src/components/Table/CollapseToggle.tsx

```diff
@@ -1,24 +1,24 @@
-import { useCallback, useContext, useState } from "react";
-import { GridCollapseContext, GridDataRow, IconButton } from "src/components";
+import { useContext } from "react";
+import { GridDataRow, IconButton, RowStateContext } from "src/components";
+import { useComputed } from "src/hooks";
 
 export interface GridTableCollapseToggleProps {
   row: GridDataRow<any>;
 }
 
 export function CollapseToggle(props: GridTableCollapseToggleProps) {
   const { row } = props;
-  const { isCollapsed, toggleCollapsed } = useContext(GridCollapseContext);
-  const [, setTick] = useState(0);
-  const currentlyCollapsed = isCollapsed(row.id);
-  const toggleOnClick = useCallback(() => {
-    toggleCollapsed(row.id);
-    setTick(Date.now());
-  }, [row.id, currentlyCollapsed, toggleCollapsed]);
-  const iconKey = currentlyCollapsed ? "chevronRight" : "chevronDown";
-  const headerIconKey = currentlyCollapsed ? "chevronsRight" : "chevronsDown";
+  const { rowState } = useContext(RowStateContext);
```

Comment:
No semantic changes, just using a `useComputed` to observe the current-collapsed state + trigger a collapse change in both us + the parent GridTable to re-calc which rows to show.